Build failure with Go v1.22/v1.23

<!-- Please keep this note for the community -->
### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!-- Thank you for keeping this note for the community -->

<!--

**Security disclosures**

If you think you’ve found a potential security issue, please do not post it in the Issues.  Instead, please email [Zinc team](mailto:admin+security@zinclabs.io).

-->

**What were you trying to achieve?**

Building the project.

**What action did you take?**

`go build ./cmd/zincsearch`

**What action/response/output did you expect?**

Successful build.

**What actually happened?**

```
go build ./cmd/zincsearch/
# github.com/pyroscope-io/godeltaprof/internal/pprof
/home/katexochen/go/pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.2/internal/pprof/delta_mutex.go:25:20: undefined: runtime_cyclesPerSecond
/home/katexochen/go/pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.2/internal/pprof/proto.go:400:8: undefined: runtime_expandFinalInlineFrame
```

**How to reproduce the issue?**

Build with Go version newer than v1.21 (which is EOL), e.g.,

```
❯ go version
go version go1.22.5 linux/amd64
```

**What version of ZincSearch are you using?**

Tried both latest release (0.4.10) and main branch (a8a490a)

**Anything else that you can tell that will help us diagnose and resolve the issue efficiently?**

/

**Any attachments/screenshots?**

/

**Which service(s) is this relate to?**

/

--- a/cmd/zincsearch/main.go
+++ b/cmd/zincsearch/main.go
@@ -28,7 +28,7 @@ import (
 
 	"github.com/getsentry/sentry-go"
 	"github.com/gin-gonic/gin"
-	"github.com/pyroscope-io/client/pyroscope"
+	"github.com/grafana/pyroscope-go"
 	"github.com/rs/zerolog/log"
 
 	"github.com/zincsearch/zincsearch/pkg/config"

--- a/go.mod
+++ b/go.mod
@@ -15,15 +15,15 @@ require (
 	github.com/gin-gonic/gin v1.9.1
 	github.com/go-ego/gse v0.80.2
 	github.com/goccy/go-json v0.10.2
+	github.com/grafana/pyroscope-go v1.2.0
 	github.com/joho/godotenv v1.4.0
 	github.com/liuzl/gocc v0.0.0-20231231122217-0372e1059ca5
 	github.com/prometheus/client_golang v1.15.1
-	github.com/pyroscope-io/client v0.6.0
 	github.com/robfig/cron/v3 v3.0.1
 	github.com/rs/zerolog v1.29.1
 	github.com/segmentio/analytics-go/v3 v3.2.1
 	github.com/shirou/gopsutil/v3 v3.23.10
-	github.com/stretchr/testify v1.8.4
+	github.com/stretchr/testify v1.9.0
 	github.com/swaggo/files v0.0.0-20220728132757-551d4a08d97a
 	github.com/swaggo/gin-swagger v1.5.3
 	github.com/swaggo/swag v1.8.12
@@ -79,9 +79,10 @@ require (
 	github.com/golang/snappy v0.0.4 // indirect
 	github.com/google/flatbuffers v1.12.1 // indirect
 	github.com/google/uuid v1.3.1 // indirect
+	github.com/grafana/pyroscope-go/godeltaprof v0.1.8 // indirect
 	github.com/josharian/intern v1.0.0 // indirect
 	github.com/json-iterator/go v1.1.12 // indirect
-	github.com/klauspost/compress v1.15.15 // indirect
+	github.com/klauspost/compress v1.17.8 // indirect
 	github.com/klauspost/cpuid/v2 v2.2.5 // indirect
 	github.com/leodido/go-urn v1.2.4 // indirect
 	github.com/liuzl/cedar-go v0.0.0-20170805034717-80a9c64b256d // indirect
@@ -101,7 +102,6 @@ require (
 	github.com/prometheus/client_model v0.3.0 // indirect
 	github.com/prometheus/common v0.42.0 // indirect
 	github.com/prometheus/procfs v0.9.0 // indirect
-	github.com/pyroscope-io/godeltaprof v0.1.2 // indirect
 	github.com/segmentio/backo-go v1.0.1 // indirect
 	github.com/shoenig/go-m1cpu v0.1.6 // indirect
 	github.com/sirupsen/logrus v1.9.3 // indirect

